22.md

```diff
@@ -0,0 +1,90 @@
+NIP-22
+======
+
+Comment
+-------
+
+`draft` `optional`
+
+A comment is meant to be used to comment on
+or reply to anything except event kinds
+whose specifications have already defined other
+ways to comment on or reply to them.
+
+For example, it can be used to comment on a [NIP-23](23.md) `kind:30023` nostr blog post
+or on an off-nostr news article or to reply to another comment.
+
+## Structure
+
+A comment is an event of `kind:1111` with plaintext `.content`
+(no HTML, no Markdown nor other formatting).
+
+It must have a `k` tag pointing to the kind of the subject being commented on.
+For example, when commenting on a `kind:30023`, the `k` tag is set to "30023".
+When replying to another `kind:1111`, the `k` tag is set to "1111".
+If commenting on an `https://abc.com/articles/1` article the `k` tag is set to "https://abc.com" (its domain).
+The `k` tag is useful to fetch top-level comments about events of specific kinds (or domains), for example,
```

Comment:
You mean like coracle that shows not only root/OP posts and top-level replies but also any-level replies in the feed?
A `K` tag with the OP kind would do it instead of using the lowercase one. I will add it.

Reply:
I consider them high signal and great to include to the feed as they carry the start of the conversation as context. Very similar to quotes.

In contrast, reply of reply of reply tend to get off-topic and carry an incomplete context.

Reply:
I can't think of any use cases for filtering by `K` and `k` tags. Maybe it's a similar discussion to #815?

Reply:
Yes, like generic reposts too. When we pick one event kind to be a generic thing, clients need a way to ask relays for just things related to a specific kind. Ex: reposts of 30023 to show on a blog's first page as "recommended articles".

Without `k`, client would have to download all generic stuff, check individually what are they referencing and discard those that reference events the client doesn't support/want.

Reply:
I quote @fiatjaf comment in #815.

> I see that reposts would cause such trouble, but reactions shouldn't as you only load them after loading the target event.

Reply:
`K` is useful if you have a reply feed, like Damus and Amethyst do. Frankly, I don't think the reply feed is that useful but so many people use it that it's kind of a requirement for kind1 feeds these days.  

And the reply feed only exists because kind1 clients couldn't download just the root threads from follows. So it was a way to justify not wasting bandwidth of the replies without showing to users.

Reply:
Ah, I see! I hadn't thought of a reply feed. Thanks Vitor.